MapboxDirections/MBRouteOptions.swift

```diff
@@ -480,19 +480,21 @@ open class RouteOptions: NSObject, NSSecureCoding, NSCopying{
     internal func response(from json: JSONDictionary) -> ([Waypoint]?, [Route]?) {
         var namedWaypoints: [Waypoint]?
         if let jsonWaypoints = (json["waypoints"] as? [JSONDictionary]) {
-            namedWaypoints = zip(jsonWaypoints, waypoints).map { (api, local) -> Waypoint in
+            namedWaypoints = zip(jsonWaypoints, self.waypoints).map { (api, local) -> Waypoint in
                 let location = api["location"] as! [Double]
                 let coordinate = CLLocationCoordinate2D(geoJSON: location)
-                return Waypoint(coordinate: coordinate, name: local.name ?? api["name"] as? String)
+                let possibleAPIName = api["name"] as? String
+                let apiName = (possibleAPIName?.isEmpty ?? true) ? nil : possibleAPIName!
```

Comment:
Core Navigation [extends String with `nonEmptyString`](https://github.com/mapbox/mapbox-navigation-ios/blob/9047a41dca57f99a08e3452a9a114dcec52f743b/MapboxCoreNavigation/String.swift#L11-L13), which is pretty convenient for situations like this. Feel free to copy it here.